### ISSUE TITLE
Create a CHS-jsPsych survey-consent plugin/template

### ISSUE THREAD

**Issue by becky-gilbert** [lookit/lookit-api#1328]
# TL;DR

We need to create a CHS-jsPsych plugin for collecting participant consent via one or more customizable survey questions.

# Narrative

As a CHS researcher, I need to run a study where I get consent from my participants using survey-style questions (rather than a video recording). I am used to using the EFP's `survey-consent` frame for this, which allows me to specify blocks of text, checkbox items, and multiple-choice questions in any order. Using the EFP version, I also don't need to do anything to implement the response validation, and the presence of my survey-consent frame is recognized as consent on the lookit-api side (i.e. responses appear in the consent ruling area).

# Acceptance Criteria

- [ ] Tests
- [ ] Documentation
- [ ] Scoping: 
  - decide how similar this plugin needs to be to the content/layout and functionality of the [EFP survey-consent frame](https://lookit.readthedocs.io/projects/frameplayer/en/latest/components/exp-lookit-survey-consent/doc.html)
  - decide whether this could be done as a template that we provide for one of the official jsPsych survey plugins, vs creating our own CHS-jsPsych plugin for this purpose
  - decide what should happen if the participant does not consent (i.e. selects 'no' rather than 'yes')
- [ ] Researcher can specify any number of text, multiple choice, and checkbox items in any order (this is dependent on scoping decisions)
- [ ] Participant must check all checkboxes and answer all multiple-choice questions to proceed (this is dependent on scoping decisions)
- [ ] Participant may not continue with the study if they do consent (which may mean checking their responses to more than one question). If they do not consent, they either remain on the page or are sent to the end of the study (dependent on scoping decisions).
- [ ] Lookit-api will recognize the presence of this trial as being consent, which means the researcher will see responses with containing this survey-consent plugin in their study's consent ruling area

# Implementation Notes

There are a number of official jsPsych plugins for creating survey-style questions. These plugins differ in the types of questions they offer (one page with a single question type for `survey-*` plugins, vs multiple question types and extra features via `survey`). They also differ in the complexity of the parameter values/configuration (the `survey` plugin is more complex because of its flexibility).
- https://www.jspsych.org/7.3/plugins/survey/
- https://www.jspsych.org/7.3/plugins/survey-html-form/
- https://www.jspsych.org/7.3/plugins/survey-likert/
- https://www.jspsych.org/7.3/plugins/survey-multi-choice/
- https://www.jspsych.org/7.3/plugins/survey-multi-select/
- https://www.jspsych.org/7.3/plugins/survey-text/

We may be able to implement this survey-consent plugin as a template for an existing/official jsPsych plugin, without creating our own version. However we do need to ensure that the lookit-api recognizes the presence of a consent response in data containing responses with this plugin, which might be tricky if we are using a vanilla jsPsych survey plugin for this.

**Comment by becky-gilbert** [comment]
FYI, I've assigned this to myself because I wrote the jsPsych survey plugin, and there are some problems with it that I need to fix. So it makes sense for me to make CHS extensions of this plugin while I'm working on it anyway.

**Comment by becky-gilbert** [comment]
Completed via https://github.com/lookit/lookit-jspsych/pull/24. 

Currently, the lookit-jsPsych survey-consent plugin produces a survey consent trial that is entirely researcher-defined. Researchers can write in their own response validation (via the `survey_function` or `validation_function` [survey parameters](https://www.jspsych.org/7.3/plugins/survey/#survey-parameters)) to check the participant's response(s) to the consent question(s), in order to determine whether they have consented and can continue with the experiment. We can provide examples in our documentation about how to do this.

At this stage, I think the additional functionality for this plugin (listed in the AC above) is out-of-scope for our first release. But we should consider adding more features in the future, e.g. to make it easier for researchers to validate consent responses and conditionally stop the experiment if the participant does not consent.
